<a href="https://colab.research.google.com/github/HeadGaur/Health-Analysis/blob/main/HealthAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install -U pandasql
!pip install --upgrade plotly==4.7.1.

  Created wheel for pandasql: filename=pandasql-0.7.3-py3-none-any.whl size=26781 sha256=af882b59f5b47f910304f0e68aef4ecedd20dd82faa8cef860599260f61d6a78
  Stored in directory: /root/.cache/pip/wheels/5c/4b/ec/41f4e116c8053c3654e2c2a47c62b4fca34cc67ef7b55deb7f
Successfully built pandasql
     |████████████████████████████████| 11.5 MB 4.8 MB/s 
  Attempting uninstall: plotly
    Found existing installation: plotly 4.4.1
    Uninstalling plotly-4.4.1:
      Successfully uninstalled plotly-4.4.1


#Importing Library for SQL

In [ ]:
import numpy as np 
import pandas as pd
import pandasql as psql
from pandasql import sqldf
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objs as go

#Importing Data

In [ ]:
users = pd.read_csv("users_data.csv")
sponsor = pd.read_csv("sponsor_data.csv")
responses = pd.read_csv("hra_responses.csv")
scores = pd.read_csv("hra_qna_scores.csv")

Checking DataFrames and Info

In [ ]:
df =  psql.sqldf("select * from users")
df.head()

id                        name  sponsor_id           created_at
0  1427541               HARSHIT TYAGI         171  2021-09-16 16:17:38
1  1427515                      VELU G         345  2021-09-16 15:56:32
2  1427509      kasturi Lakshmi Naresh         405  2021-09-16 15:49:11
3  1427465  Sandeep Kumar Nagullapally         405  2021-09-16 15:08:12
4  1427458      AMAR PREET SINGH JAGGI         171  2021-09-16 14:59:13

In [ ]:
spdf =  psql.sqldf("select * from sponsor")
df.head()

id                        name  sponsor_id           created_at
0  1427541               HARSHIT TYAGI         171  2021-09-16 16:17:38
1  1427515                      VELU G         345  2021-09-16 15:56:32
2  1427509      kasturi Lakshmi Naresh         405  2021-09-16 15:49:11
3  1427465  Sandeep Kumar Nagullapally         405  2021-09-16 15:08:12
4  1427458      AMAR PREET SINGH JAGGI         171  2021-09-16 14:59:13

In [ ]:
df =  psql.sqldf("select * from responses")
responses.head()

id  user_id  question_id response           created_at
0  2508505  1407131           40    start  2021-08-16 01:15:14
1  2508506  1407131            1   Female  2021-08-16 01:15:18
2  2508507  1407131            2       32  2021-08-16 01:15:23
3  2508508  1407131            3       65  2021-08-16 01:15:27
4  2508509  1407131            4      157  2021-08-16 01:15:35

In [ ]:
scores.head()

question_id           type  ... options score
0            1  single_select  ...    Male   NaN
1            1  single_select  ...  Female   NaN
2            2     text_input  ...     NaN   NaN
3            3     text_input  ...     NaN   NaN
4            4     text_input  ...     NaN   NaN

[5 rows x 6 columns]

#Joined Response and Scores tables

In [ ]:
SRdf = psql.sqldf("select scores.type,scores.title,scores.text,scores.options,scores.score,scores.question_id, responses.id as response_ID,responses.user_id,responses.response,responses.created_at from scores join responses on scores.question_id=responses.question_id;")
SRdf.isnull().sum()

type               0
title              0
text               0
options         3646
score          12368
question_id        0
response_ID        0
user_id            0
response           0
created_at         0
dtype: int64

#ScoreResponse Table Join User data Table

In [ ]:
df =  psql.sqldf("select SRdf.type, SRdf.title, SRdf.text, SRdf.options, SRdf.score, SRdf.question_id, SRdf.response_ID,SRdf.response, users.id as user_id, users.name,users.sponsor_id,users.created_at from SRdf join users on users.id=SRdf.user_id;")
df.head(n=10)

type          title  ... sponsor_id           created_at
0  single_select  Basic Profile  ...         75  2021-08-16 00:51:02
1  single_select  Basic Profile  ...        345  2021-08-16 06:50:27
2  single_select  Basic Profile  ...        171  2021-08-16 08:59:31
3  single_select  Basic Profile  ...        345  2021-08-16 09:02:01
4  single_select  Basic Profile  ...        345  2021-08-16 09:54:11
5  single_select  Basic Profile  ...         75  2021-08-16 09:49:16
6  single_select  Basic Profile  ...        345  2021-08-16 10:25:45
7  single_select  Basic Profile  ...        345  2021-08-16 11:17:20
8  single_select  Basic Profile  ...        345  2021-08-16 11:24:31
9  single_select  Basic Profile  ...        345  2021-08-16 11:42:03

[10 rows x 12 columns]

In [ ]:
df =  psql.sqldf("select df.type, df.title, df.text, df.options, df.score, df.question_id, df.response_ID, df.response, df.user_id, df.name,df.sponsor_id,df.created_at,sponsor.name as sponsor_name from df join sponsor on sponsor.id=df.sponsor_id;")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47927 entries, 0 to 47926
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   type          47927 non-null  object 
 1   title         47927 non-null  object 
 2   text          47927 non-null  object 
 3   options       44281 non-null  object 
 4   score         35559 non-null  float64
 5   question_id   47927 non-null  int64  
 6   response_ID   47927 non-null  int64  
 7   response      47927 non-null  object 
 8   user_id       47927 non-null  int64  
 9   name          47927 non-null  object 
 10  sponsor_id    47927 non-null  int64  
 11  created_at    47927 non-null  object 
 12  sponsor_name  47927 non-null  object 
dtypes: float64(1), int64(4), object(8)
memory usage: 4.8+ MB


In [ ]:
df.head(n=10)

type          title  ...           created_at sponsor_name
0  single_select  Basic Profile  ...  2021-08-16 00:51:02   Sponsor 75
1  single_select  Basic Profile  ...  2021-08-16 06:50:27  Sponsor 345
2  single_select  Basic Profile  ...  2021-08-16 08:59:31  Sponsor 171
3  single_select  Basic Profile  ...  2021-08-16 09:02:01  Sponsor 345
4  single_select  Basic Profile  ...  2021-08-16 09:54:11  Sponsor 345
5  single_select  Basic Profile  ...  2021-08-16 09:49:16   Sponsor 75
6  single_select  Basic Profile  ...  2021-08-16 10:25:45  Sponsor 345
7  single_select  Basic Profile  ...  2021-08-16 11:17:20  Sponsor 345
8  single_select  Basic Profile  ...  2021-08-16 11:24:31  Sponsor 345
9  single_select  Basic Profile  ...  2021-08-16 11:42:03  Sponsor 345

[10 rows x 13 columns]

#Types of Questions


In [ ]:
Questiondf =  psql.sqldf("select distinct question_id, text from df; ")
Questiondf

question_id                                               text
0             1                         Please mention your gender
1             2                              Please enter your age
2             3                   Please enter you weight (in kgs)
3             4                  Please enter your height (in cms)
4             5             How do you describe your work profile?
5             6  On average, how much time do you spend on phys...
6             7                              Do you feel stressed?
7             8      Have you got your cholesterol profiling done?
8             9  What are your total cholesterol levels (in mg/dL)
9            10          Please mention your HDL levels (in mg/dL)
10           11                   What are your recent LDL levels?
11           12          What are your latest Triglyceride levels?
12           13                                      Do you smoke?
13           14         How many cigarettes do you smoke in a day?
14           15  What describes your intention to stop smoking ...
15           16  Have you got your Blood Pressure checked anytime?
16           17      How often do you measure your Blood Pressure?
17           18  Please mention the latest reading of your Bloo...
18           19           Are you on treatment for Blood Pressure?
19           20            Do you check your fasting sugar levels?
20           21    What are your usual Fasting Blood Sugar levels?
21           22                               Do you have Diabetes
22           23                 What type of diabetes do you have?
23           24                                What is your HbA1C?
24           25         Are you on treatment for high blood sugar?
25           26  Do you have family history of any of the follo...
26           27                     Do you have any thyroid issue?
27           28                    What are your latest TSH levels
28           29                What are your latest Free T3 levels
29           30                What are your latest Free T4 levels
30           31                   What is your daily water intake?
31           32  How many days in a week do you have a sound sl...
32           33  Has your sleep been affected by any worry lately?
33           34      On the whole, how well have you been feeling?
34           35                            Do you consume alcohol?
35           36                  How much alcohol do you consume? 
36           37  Have you ever felt you should cut down on your...
37           38  Have you ever felt bad or guilty about your dr...
38           39  How often do you drink sugary beverages? (Exam...
39           40                                          HRA Start

# Score of Users 

Here, responses dataframe and users dataframe had to joined as left join , but due to inner join we lost some data. 

++few users are showing as score ~ 300+ thats wrong, need to remove duplicates entries here.

In [ ]:
user_score_df =  psql.sqldf("select user_id,name,sum(score) as score from df group by user_id")
user_score_df.head()

user_id            name  score
0  1407097  Prudhvi Geddam    NaN
1  1407131     Swara Mehta  131.0
2  1407173           Vinod    NaN
3  1407188         Mohan G   81.0
4  1407216       Yathish S  108.0

### **Creating Height and Weight Columns**

In [ ]:
age_height_weight_df = psql.sqldf("SELECT user_id,name,SUM(CASE WHEN question_id = 3 THEN response ELSE 0 END) AS Weight,SUM(CASE WHEN question_id = 4 THEN response ELSE 0 END) AS Height FROM df GROUP BY user_id;")
age_height_weight_df.head(n=5)

user_id            name  Weight  Height
0  1407097  Prudhvi Geddam     0.0     0.0
1  1407131     Swara Mehta    65.0   157.0
2  1407173           Vinod     0.0     0.0
3  1407188         Mohan G    80.0   180.0
4  1407216       Yathish S    84.0   171.0

### **Creating Gender Column**

In [ ]:
gender_df = psql.sqldf("SELECT user_id,(CASE WHEN question_id = 1 THEN response ELSE 0 END) AS Gender FROM df WHERE question_id = 1 GROUP BY user_id;")
gender_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 950 entries, 0 to 949
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   user_id  950 non-null    int64 
 1   Gender   950 non-null    object
dtypes: int64(1), object(1)
memory usage: 15.0+ KB


In [ ]:
ahwg_df = pd.merge(age_height_weight_df, gender_df, on = 'user_id')
ahwg_df.head(n=5)

user_id            name  Weight  Height  Gender
0  1407097  Prudhvi Geddam     0.0     0.0    Male
1  1407131     Swara Mehta    65.0   157.0  Female
2  1407173           Vinod     0.0     0.0    Male
3  1407188         Mohan G    80.0   180.0    Male
4  1407216       Yathish S    84.0   171.0    Male

#Gender Distribution

In [ ]:
fig = px.treemap(ahwg_df, path=['Gender'], color='Gender')
fig.update_layout(margin = dict(t=60, l=15, r=15, b=15),
                  title_text="<b>Gender Distribution</b>",
                  title_x=0.5,
                  font=dict(family="serif", size=20, color='rgb(5, 14, 48)'))
fig.show()

#Smoking Category

In [ ]:
smoke_df_k = psql.sqldf("SELECT user_id,(CASE WHEN question_id = 14 THEN response ELSE 0 END) AS 'No_of_Cigarette' FROM df WHERE question_id = 14 GROUP BY user_id;")
smoke_df_count = psql.sqldf("select No_of_cigarette, count(No_of_cigarette) as count from smoke_df_k group by No_of_cigarette")
smoke_df_count

No_of_Cigarette  count
0            5-10     36
1     Less than 5    115
2    More than 10     13

#Distribution of Chain-smokers

In [ ]:

tmp_smoke = smoke_df_k.No_of_Cigarette.value_counts()
smoke_labels = (np.array(tmp_smoke.index))
smoke_sizes = (np.array((tmp_smoke / tmp_smoke.sum())*100))


fig = go.Figure(data=[go.Pie(labels=smoke_labels, values=smoke_sizes, hole=.5)])
fig.update_layout( title_text="Smoke Distribution",
    annotations=[dict(text='🚬', x=0.5, y=0.52, font_size=60, showarrow=False)])

#Sponsor Data

In [ ]:
sponsor_df = psql.sqldf("select sponsor.name as sponsor_name, users.id from sponsor join users on sponsor.id=users.sponsor_id;")
sponsor_count = psql.sqldf("select sponsor_name,count(sponsor_name) as Count from sponsor_df group by sponsor_name;")
fig = px.bar(sponsor_count, x='sponsor_name', y='Count')
fig.show()

In [ ]:
score_df = psql.sqldf("select name, sum(score) from df group by user_id;")
score_df.head()

name  sum(score)
0  Prudhvi Geddam         NaN
1     Swara Mehta       131.0
2           Vinod         NaN
3         Mohan G        81.0
4       Yathish S       108.0

Score-user-linechart


In [ ]:

fig = px.line(user_score_df, x="name", y="score", title='User Score')
fig.show()

In [ ]:
score_df = psql.sqldf("select distinct response,user_id,response_ID from df where name = 'Nilesh Suryawanshi' order by user_id;")
score_df

response  user_id  response_ID
0                                Male  1415214      2536758
1                                  42  1415214      2536759
2                                  76  1415214      2536760
3                                187   1415214      2536761
4                            Desk job  1415214      2536762
5   Light exercise: 1-3 days per week  1415214      2536763
6                        Occasionally  1415214      2536764
7                                 Yes  1415214      2536765
8                             160-199  1415214      2536766
9                               40-49  1415214      2536767
10                            130-159  1415214      2536768
11                            150-200  1415214      2536771
12                                 No  1415214      2536772
13                                Yes  1415214      2536777
14                       Once a month  1415214      2536780
15                     130-139/ 85-90  1415214      2536783
16                                 No  1415214      2536786
17                                Yes  1415214      2536788
18                            100-125  1415214      2536791
19                                Yes  1415214      2536795
20                             Type 2  1415214      2536832
21                            6.0-6.5  1415214      2536833
22                                 No  1415214      2536834
23                       Hypertension  1415214      2536837
24                                 No  1415214      2536839
25                More than 8 glasses  1415214      2536841
26                   More than 3 days  1415214      2536844
27                 No more than usual  1415214      2536849
28             Happy most of the time  1415214      2536855
29                                Yes  1415214      2536857
30                  Occasional Intake  1415214      2536858
31                                Yes  1415214      2536861
32                                Yes  1415214      2536862
33                 0-2 times per week  1415214      2536865
34                              start  1415214      2536757

In [ ]:
diabeties_df = psql.sqldf("SELECT user_id,name,SUM(CASE WHEN question_id = 20 THEN response ELSE 0 END) AS fasting_sugar_levels_check,SUM(CASE WHEN question_id = 21 THEN response ELSE 0 END) AS fasting_sugar_levels FROM df GROUP BY user_id;")
age_height_weight_df.head(n=5)

user_id            name  Weight  Height
0  1407097  Prudhvi Geddam     0.0     0.0
1  1407131     Swara Mehta    65.0   157.0
2  1407173           Vinod     0.0     0.0
3  1407188         Mohan G    80.0   180.0
4  1407216       Yathish S    84.0   171.0

In [ ]:
kdf = psql.sqldf("delete t1 from df t1 inner join df as t2 where t1.response_ID < t2.response_ID and t1.user_id=t2.user_id and t1.question_id=t2.question_id;")

PandaSQLException: ignored

In [ ]:
responses = psql.sqldf("SELECT id, user_id, question_id, response,DATE(created_at) as date from responses;")
responses

users = psql.sqldf("SELECT id,name, sponsor_id,DATE(created_at) as date from users;")
users



id                        name  sponsor_id        date
0    1427541               HARSHIT TYAGI         171  2021-09-16
1    1427515                      VELU G         345  2021-09-16
2    1427509      kasturi Lakshmi Naresh         405  2021-09-16
3    1427465  Sandeep Kumar Nagullapally         405  2021-09-16
4    1427458      AMAR PREET SINGH JAGGI         171  2021-09-16
..       ...                         ...         ...         ...
995  1407216                   Yathish S         171  2021-08-16
996  1407188                     Mohan G         345  2021-08-16
997  1407173                       Vinod         345  2021-08-16
998  1407131                 Swara Mehta          75  2021-08-16
999  1407097              Prudhvi Geddam         220  2021-08-15

[1000 rows x 4 columns]

In [ ]:
response_user_df = psql.sqldf("SELECT users.name, users.sponsor_id,users.date, responses.id as response_ID,responses.user_id,responses.response from responses left join users on responses.user_id = users.id;")
response_user_df

name  sponsor_id  ...  user_id                response
0                 Swara Mehta          75  ...  1407131                   start
1                 Swara Mehta          75  ...  1407131                  Female
2                 Swara Mehta          75  ...  1407131                      32
3                 Swara Mehta          75  ...  1407131                      65
4                 Swara Mehta          75  ...  1407131                     157
...                       ...         ...  ...      ...                     ...
18954  kasturi Lakshmi Naresh         405  ...  1427509     Between 5-8 glasses
18955  kasturi Lakshmi Naresh         405  ...  1427509        More than 3 days
18956  kasturi Lakshmi Naresh         405  ...  1427509  Rather more than usual
18957  kasturi Lakshmi Naresh         405  ...  1427509  Happy most of the time
18958  kasturi Lakshmi Naresh         405  ...  1427509                      No

[18959 rows x 6 columns]

In [ ]:
df.head(n=2)

type          title  ...           created_at sponsor_name
0  single_select  Basic Profile  ...  2021-08-16 00:51:02   Sponsor 75
1  single_select  Basic Profile  ...  2021-08-16 06:50:27  Sponsor 345

[2 rows x 13 columns]

In [ ]:
sq = psql.sqldf("SELECT distinct user_id,question_id,score,response, response_id,name from df;")
sq

user_id  question_id  ...  response_ID                        name
0      1407131            1  ...      2508506                 Swara Mehta
1      1407188            1  ...      2508563                     Mohan G
2      1407216            1  ...      2508707                   Yathish S
3      1407218            1  ...      2508730            J V John Abinesh
4      1407249            1  ...      2508838                Sahid Perwez
...        ...          ...  ...          ...                         ...
35157  1425839           40  ...      2553655                      Naresh
35158  1427465           40  ...      2553657  Sandeep Kumar Nagullapally
35159  1427515           40  ...      2553661                      VELU G
35160  1427541           40  ...      2553685               HARSHIT TYAGI
35161  1427509           40  ...      2553690      kasturi Lakshmi Naresh

[35162 rows x 6 columns]

In [ ]:
_df = psql.sqldf("select response,user_id,response_ID from response_user_df where name = 'Nilesh Suryawanshi' order by user_id;")
_df

response  user_id  response_ID
0                               start  1415214      2536757
1                                Male  1415214      2536758
2                                  42  1415214      2536759
3                                  76  1415214      2536760
4                                187   1415214      2536761
5                            Desk job  1415214      2536762
6   Light exercise: 1-3 days per week  1415214      2536763
7                        Occasionally  1415214      2536764
8                                 Yes  1415214      2536765
9                             160-199  1415214      2536766
10                              40-49  1415214      2536767
11                            130-159  1415214      2536768
12                            150-200  1415214      2536771
13                                 No  1415214      2536772
14                                Yes  1415214      2536777
15                       Once a month  1415214      2536780
16                     130-139/ 85-90  1415214      2536783
17                                 No  1415214      2536786
18                                Yes  1415214      2536788
19                            100-125  1415214      2536791
20                                Yes  1415214      2536795
21                             Type 2  1415214      2536832
22                            6.0-6.5  1415214      2536833
23                                 No  1415214      2536834
24                       Hypertension  1415214      2536837
25                                 No  1415214      2536839
26                More than 8 glasses  1415214      2536841
27                   More than 3 days  1415214      2536844
28                 No more than usual  1415214      2536849
29             Happy most of the time  1415214      2536855
30                                Yes  1415214      2536857
31                  Occasional Intake  1415214      2536858
32                                Yes  1415214      2536861
33                                Yes  1415214      2536862
34                 0-2 times per week  1415214      2536865

In [ ]:
response_user_df.head(n=2)

name  sponsor_id        date  response_ID  user_id response
0  Swara Mehta          75  2021-08-16      2508505  1407131    start
1  Swara Mehta          75  2021-08-16      2508506  1407131   Female

In [ ]:
df

type          title  ...           created_at sponsor_name
0      single_select  Basic Profile  ...  2021-08-16 00:51:02   Sponsor 75
1      single_select  Basic Profile  ...  2021-08-16 06:50:27  Sponsor 345
2      single_select  Basic Profile  ...  2021-08-16 08:59:31  Sponsor 171
3      single_select  Basic Profile  ...  2021-08-16 09:02:01  Sponsor 345
4      single_select  Basic Profile  ...  2021-08-16 09:54:11  Sponsor 345
...              ...            ...  ...                  ...          ...
47922     text_input      Start HRA  ...  2021-09-13 11:19:28  Sponsor 405
47923     text_input      Start HRA  ...  2021-09-16 15:08:12  Sponsor 405
47924     text_input      Start HRA  ...  2021-09-16 15:56:32  Sponsor 345
47925     text_input      Start HRA  ...  2021-09-16 16:17:38  Sponsor 171
47926     text_input      Start HRA  ...  2021-09-16 15:49:11  Sponsor 405

[47927 rows x 13 columns]

In [ ]:
diabeties_df = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 20 THEN response ELSE 0 END) AS fasting_sugar_levels_check FROM df where question_id = 20 GROUP BY user_id;")
diabeties_df["fasting_sugar_levels_check"].value_counts()

No     580
Yes    151
Name: fasting_sugar_levels_check, dtype: int64

In [ ]:
diabeties_level_df = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 21 THEN response ELSE 0 END) AS fasting_sugar_levels FROM df where question_id = 21 GROUP BY user_id;")
diabeties_level_df

user_id                    name fasting_sugar_levels
0    1407272  Nitin Narayanan Baoker                70-99
1    1407431           SHAHRUKH KHAN        Lower than 70
2    1407449    Vaishnavi Narasimhan              100-125
3    1407830         Abhishek Sharma                70-99
4    1407898          surya v singh                 70-99
..       ...                     ...                  ...
145  1426822             TARUN GHATA                70-99
146  1426832    chandi Charan Mandal                70-99
147  1427110           khaleel Ahmed        Lower than 70
148  1427144         LAMBODAR MISHRA                70-99
149  1427167             ARUN ANITHA        More than 125

[150 rows x 3 columns]

In [ ]:
diabeties_merge_df = psql.sqldf("SELECT diabeties_level_df.user_id,diabeties_level_df.name,diabeties_level_df.fasting_sugar_levels,diabeties_df.fasting_sugar_levels_check from diabeties_level_df join diabeties_df on diabeties_level_df.user_id=diabeties_df.user_id;")
diabeties_merge_df

user_id  ... fasting_sugar_levels_check
0    1407272  ...                        Yes
1    1407431  ...                        Yes
2    1407449  ...                        Yes
3    1407830  ...                        Yes
4    1407898  ...                        Yes
..       ...  ...                        ...
145  1426822  ...                        Yes
146  1426832  ...                        Yes
147  1427110  ...                        Yes
148  1427144  ...                        Yes
149  1427167  ...                        Yes

[150 rows x 4 columns]

In [ ]:
diabeties_yes_no = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 22 THEN response ELSE 0 END) AS IsSugar FROM df where question_id = 22 GROUP BY user_id;")
diabeties_yes_no["IsSugar"].value_counts()

No     707
Yes     23
Name: IsSugar, dtype: int64

In [ ]:
diabeties = psql.sqldf("SELECT diabeties_merge_df.user_id,diabeties_merge_df.name,diabeties_merge_df.fasting_sugar_levels,diabeties_merge_df.fasting_sugar_levels_check,diabeties_yes_no.IsSugar from diabeties_merge_df join diabeties_yes_no on diabeties_merge_df.user_id=diabeties_yes_no.user_id;")
diabeties

user_id                    name  ... fasting_sugar_levels_check IsSugar
0    1407272  Nitin Narayanan Baoker  ...                        Yes      No
1    1407431           SHAHRUKH KHAN  ...                        Yes      No
2    1407449    Vaishnavi Narasimhan  ...                        Yes      No
3    1407830         Abhishek Sharma  ...                        Yes      No
4    1407898          surya v singh   ...                        Yes      No
..       ...                     ...  ...                        ...     ...
145  1426822             TARUN GHATA  ...                        Yes      No
146  1426832    chandi Charan Mandal  ...                        Yes      No
147  1427110           khaleel Ahmed  ...                        Yes      No
148  1427144         LAMBODAR MISHRA  ...                        Yes      No
149  1427167             ARUN ANITHA  ...                        Yes     Yes

[150 rows x 5 columns]

In [ ]:
smoke_yes_no = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 13 THEN response ELSE 0 END) AS IsSmoke FROM df where question_id = 13 GROUP BY user_id;")
smoke_yes_no["IsSmoke"].value_counts()

No     582
Yes    163
Name: IsSmoke, dtype: int64

In [ ]:
Alcohal_yes_no = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 35 THEN response ELSE 0 END) AS IsAlcohal FROM df where question_id = 35 GROUP BY user_id;")
Alcohal_yes_no["IsAlcohal"].value_counts()

No     473
Yes    227
Name: IsAlcohal, dtype: int64

In [ ]:
sugary_bev = psql.sqldf("SELECT user_id,name,(CASE WHEN question_id = 39 THEN response ELSE 0 END) AS Sugarybev FROM df where question_id = 39 GROUP BY user_id;")
sugary_bev["Sugarybev"].value_counts()

0-2 times per week          127
7 or more times per week     63
3-6 times per week           35
Name: Sugarybev, dtype: int64

In [ ]:
smoke = psql.sqldf("SELECT diabeties.user_id,diabeties.name,diabeties.fasting_sugar_levels,diabeties.fasting_sugar_levels_check,diabeties.IsSugar,smoke_yes_no.IsSmoke from diabeties left join smoke_yes_no on diabeties.user_id=smoke_yes_no.user_id;")
smoke

user_id                    name  ... IsSugar IsSmoke
0    1407272  Nitin Narayanan Baoker  ...      No      No
1    1407431           SHAHRUKH KHAN  ...      No     Yes
2    1407449    Vaishnavi Narasimhan  ...      No      No
3    1407830         Abhishek Sharma  ...      No      No
4    1407898          surya v singh   ...      No     Yes
..       ...                     ...  ...     ...     ...
145  1426822             TARUN GHATA  ...      No      No
146  1426832    chandi Charan Mandal  ...      No      No
147  1427110           khaleel Ahmed  ...      No      No
148  1427144         LAMBODAR MISHRA  ...      No      No
149  1427167             ARUN ANITHA  ...     Yes      No

[150 rows x 6 columns]

In [ ]:
Alcohal = psql.sqldf("SELECT t1.user_id,t1.name,t1.fasting_sugar_levels,t1.fasting_sugar_levels_check,t1.IsSugar,t1.IsSmoke,t2.IsAlcohal from smoke as t1 left join Alcohal_yes_no as t2 on t1.user_id=t2.user_id;")
Alcohal

user_id                    name  ... IsSmoke IsAlcohal
0    1407272  Nitin Narayanan Baoker  ...      No       Yes
1    1407431           SHAHRUKH KHAN  ...     Yes      None
2    1407449    Vaishnavi Narasimhan  ...      No        No
3    1407830         Abhishek Sharma  ...      No        No
4    1407898          surya v singh   ...     Yes       Yes
..       ...                     ...  ...     ...       ...
145  1426822             TARUN GHATA  ...      No        No
146  1426832    chandi Charan Mandal  ...      No        No
147  1427110           khaleel Ahmed  ...      No        No
148  1427144         LAMBODAR MISHRA  ...      No       Yes
149  1427167             ARUN ANITHA  ...      No        No

[150 rows x 7 columns]

In [ ]:
Sugary_df = psql.sqldf("SELECT t1.user_id,t1.name,t1.fasting_sugar_levels,t1.fasting_sugar_levels_check,t1.IsSugar,t1.IsSmoke,t1.IsAlcohal,t2.Sugarybev from Alcohal as t1 left join sugary_bev as t2 on t1.user_id=t2.user_id;")
Sugary_df

user_id                    name  ... IsAlcohal                 Sugarybev
0    1407272  Nitin Narayanan Baoker  ...       Yes        0-2 times per week
1    1407431           SHAHRUKH KHAN  ...      None                      None
2    1407449    Vaishnavi Narasimhan  ...        No                      None
3    1407830         Abhishek Sharma  ...        No                      None
4    1407898          surya v singh   ...       Yes  7 or more times per week
..       ...                     ...  ...       ...                       ...
145  1426822             TARUN GHATA  ...        No                      None
146  1426832    chandi Charan Mandal  ...        No                      None
147  1427110           khaleel Ahmed  ...        No                      None
148  1427144         LAMBODAR MISHRA  ...       Yes        0-2 times per week
149  1427167             ARUN ANITHA  ...        No                      None

[150 rows x 8 columns]

  #  

#  

#  

# Predicting Diabetes Patients

using Features as DoTheySmoke, DoTheyDrink, HowManyTimesColdDrinkInAWeek etc.

In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import RobustScaler
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import LabelEncoder

In [ ]:
x = Sugary_df.drop(['IsSugar'],axis=1)
y = Sugary_df['IsSugar']

In [ ]:
# import preprocessing from sklearn
from sklearn import preprocessing
le = preprocessing.LabelEncoder()

x = x.apply(le.fit_transform)
x.head()

user_id  name  fasting_sugar_levels  ...  IsSmoke  IsAlcohal  Sugarybev
0        0    83                     1  ...        0          1          0
1        1   108                     2  ...        1          2          3
2        2   130                     0  ...        0          0          3
3        3     9                     1  ...        0          0          3
4        4   148                     1  ...        1          1          2

[5 rows x 7 columns]

# Out of 149 , separated 
#38 Rows to test the model
#111 Rows to train the model

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size=0.25, random_state=0)

In [ ]:
cols = x_train.columns

x_train = pd.DataFrame(x_train,columns = cols)
x_test = pd.DataFrame(x_test,columns = cols)
x_train

user_id  name  fasting_sugar_levels  ...  IsSmoke  IsAlcohal  Sugarybev
61        61   106                     1  ...        0          0          3
92        92    16                     1  ...        1          1          0
112      112   122                     2  ...        0          0          3
2          2   130                     0  ...        0          0          3
141      141   127                     1  ...        0          1          0
..       ...   ...                   ...  ...      ...        ...        ...
9          9    19                     1  ...        1          1          2
103      103   132                     2  ...        0          0          3
67        67   133                     1  ...        0          1          0
117      117    42                     0  ...        0          0          3
47        47   103                     1  ...        0          1          1

[112 rows x 7 columns]

In [ ]:
LR = LogisticRegression(solver='liblinear')
LR.fit(x_train,y_train)

LogisticRegression(solver='liblinear')

In [ ]:
y_pred = LR.predict(x_test)
y_pred

array(['No', 'No', 'No', 'No', 'No', 'No', 'Yes', 'No', 'No', 'No', 'No',
       'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No',
       'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No', 'No',
       'No', 'No', 'No', 'No', 'No'], dtype=object)

In [ ]:
from sklearn.metrics import confusion_matrix
confusion_matrix(y_test,y_pred)

array([[28,  0],
       [ 9,  1]])

As Result shown - 
28 were shown True-Positive , means prediction matched for having no diabetes - 

9 were shown False-Negative : Type 2 error

1 as True Negative 